DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/WKWebViewRecorder.swift

```diff
@@ -12,42 +12,61 @@ internal class WKWebViewRecorder: NodeRecorder {
     let identifier = UUID()
 
     func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) -> NodeSemantics? {
-        guard let webView = view as? WKWebView else {
+        guard let webview = view as? WKWebView else {
             return nil
         }
 
-        guard attributes.isVisible else {
-            return InvisibleElement.constant
-        }
-
-        let builder = WKWebViewWireframesBuilder(
-            wireframeID: context.ids.nodeID(view: view, nodeRecorder: self),
-            slotID: webView.configuration.userContentController.hash,
-            attributes: attributes
-        )
+        // Record all webviews regardless of their visibility
+        let slot = WKWebViewSlot(webview: webview)
+        // Add or update the webview slot in cache
+        context.webviewCache.update(slot)
 
+        let builder = WKWebViewWireframesBuilder(slot: slot, attributes: attributes)
         let node = Node(viewAttributes: attributes, wireframesBuilder: builder)
         return SpecificElement(subtreeStrategy: .ignore, nodes: [node])
     }
 }
 
+/// The slot recorded for a `WKWebView`.
+internal struct WKWebViewSlot: WebViewSlot {
```

Comment:
I think it can be potentially solved with NSHashTable:
https://developer.apple.com/documentation/foundation/nshashtable

`NSHashTable<WKWebview>(options: .weakMemory)`

Should ease the implementation as it's relying on NSObject's hash internally if I'm not mistaken.

There's also [NSMapTable](https://developer.apple.com/documentation/foundation/nsmaptable) if we want to control keys.

Reply:
That's a good call! It indeed simplify things, thanks for the suggestion.